ocean_lib/models/v4/test/test_factory_router.py

```diff
@@ -0,0 +1,444 @@
+        FactoryRouter.EVENT_NEW_POOL,
+        tx_receipt.blockNumber,
+        web3.eth.block_number,
+        None,
+    )
+    pool1 = registered_event_pool[0]["args"]["poolAddress"]
+
+    nftData2 = {
+        "name": "72120Bundle",
+        "symbol": "72Bundle",
+        "templateIndex": 1,
+        "baseURI": "https://oceanprotocol.com/nft2/",
+    }
+
+    ercData2 = {
+        "templateIndex": 1,
+        "strings": ["ERC20B12", "ERC20DT1Symbol2"],
+        "addresses": [
+            factory_deployer.address,
+            consumer_wallet.address,
+            factory_deployer.address,
+            ZERO_ADDRESS,
+        ],
+        "uints": [web3.toWei("1000000", "ether"), 0],
+        "bytess": [],
+    }
+
+    poolData2 = {
+        "addresses": [
+            v4_addresses["Staking"],
+            v4_addresses["Ocean"],
+            v4_addresses["ERC721Factory"],
+            factory_deployer.address,
+            factory_deployer.address,
+            v4_addresses["poolTemplate"],
+        ],
+        "ssParams": [
+            web3.toWei("1", "ether"),
+            oceanContract.decimals(),
+            web3.toWei("10000", "ether"),
+            2500000,
+            web3.toWei("2", "ether"),
+        ],
+        "swapFees": [web3.toWei("0.001", "ether"), web3.toWei("0.001", "ether")],
+    }
+
+    tx = nftFactory.create_nft_erc_with_pool(
+        nftData2, ercData2, poolData2, factory_deployer
+    )
+
+    tx_receipt = web3.eth.wait_for_transaction_receipt(tx)
+    registered_event = nftFactory.get_event_log(
+        "TokenCreated", tx_receipt.blockNumber, web3.eth.block_number, None
+    )
+    ercToken2 = registered_event[0]["args"]["newTokenAddress"]
+    ercTokenContract2 = ERC20Token(web3=web3, address=ercToken2)
+    registered_event_pool = ercTokenContract2.get_event_log(
+        FactoryRouter.EVENT_NEW_POOL,
+        tx_receipt.blockNumber,
+        web3.eth.block_number,
+        None,
+    )
+    pool2 = registered_event_pool[0]["args"]["poolAddress"]
+
+    op1 = {
+        "exchangeIds": Web3.keccak(0x00),
+        "source": pool1,
+        "operation": 0,
+        "tokenIn": v4_addresses["Ocean"],
+        "amountsIn": web3.toWei("1", "ether"),
+        "tokenOut": ercToken,
+        "amountsOut": web3.toWei("0.1", "ether"),
+        "maxPrice": web3.toWei("10", "ether"),
+    }
+
+    op2 = {
+        "exchangeIds": Web3.keccak(0x00),
+        "source": pool2,
+        "operation": 0,
+        "tokenIn": v4_addresses["Ocean"],
+        "amountsIn": web3.toWei("1", "ether"),
+        "tokenOut": ercToken2,
+        "amountsOut": web3.toWei("0.1", "ether"),
+        "maxPrice": web3.toWei("10", "ether"),
+    }
+    ercTokenContract.approve(
+        v4_addresses["ERC721Factory"], 2 ** 256 - 1, factory_deployer
+    )
+    ercTokenContract2.approve(
+        v4_addresses["ERC721Factory"], 2 ** 256 - 1, factory_deployer
+    )
+
+    balanceDaiBefore = oceanContract.balanceOf(factory_deployer.address)
```

Comment:
Remove DAI occurrences as variables declarations when you work with Ocean tokens.